### ISSUE TITLE
release 2018-02 follow-up

### ISSUE THREAD

**Issue by stonebig** [winpython/winpython#609]
expected (new) date : july 8th

The moving part closely followed of this release is Python-3.7 with numba.
The other parts are in follow-up mode. (10x less effort)

expected features:
 - WinPython-3.7.0.1
 - numpy-1.14.4, scipy-1.1.0 , pandas-0.23.1, statsmodels-0.9.0
 - Holoviews-1.10.5+, Bokeh-0.12.16+, Matplolib-2.2.2, hvplot (simplifying layer over holoviews)
 - Jupyterlab-0.32.1+, Altair-2.1.0
 - PySide2-5.12.0preview support (works with pyzo/qtconsole/pyqtgraph, not spyder etc...)

sustained effort of this release: 
- WinPython-3.7.0.1, (droping the "Qt5" tag: no more Qt4),
- PySide2-5.11.0dev survival experience:
   . QtConsole: ok,
   . pyzo: ok ,
   . designer.exe: ok,
   . Spyder: wiil need a summer of adapation. 
- numeric stack refresh.
- explore DL / Pymc4 / Keras dimensions : PyTorch / chainer / CNTK  / nothing 
   . we are forced to experiment switch on short term by technical constraint: size + old cpu,
   . build1 = Pytorch,
   . build2 = Chainer + Cupy ? or cntk that has shrink to 8Mo ???

reduced effort: (follower mode)
- PyQt5 stack (will stick on stable 8.9.x, the stable branch, next stable is 5.12.x)
- Jupyterlab

**Comment by stonebig** [comment]
audit-trail:

b1

b2

rc

**Comment by stonebig** [comment]
build a1 (for alpha of PySide2 integration)

This build is intended for users willing to help maturation of  PySide2.

Changes from WinPython 2018-01:
- numpy-1.14.3, jupyterlab-0.32.1 (beta2.1)
- alpha support for PySide2-5.11a1:
  . having to delete QML part so it fits in the installer, (uninstall pip install it for a full experience),
  .  QtConsole Works, WPPM works, Pyzo works expect crashes at closing (a PySide2 bug),
  . the rest is blowing up: spyder, PyQtgraph, 
- adding pyzo editor, as Spyder doesn't support PySide2 yet.


Areas of interest for testers:
- does pyzo work ok ?  (for users who knows it well)
- participating to the great PySide2 integration in Qt-dependant packages:
    . propose patches to PyQtGraph, Spyder, Guiqwt, Guidata, Rx, ...
    . check QtDesigner works well,
- numpy-1.14.3 integration.

**Comment by RoyiAvital** [comment]
PyTorch 0.4 is out with support for Windows.
It is really the best out there (It is also like a GPU accelerated Numpy) and I think you should include it and drop other DL to save space.

**Comment by stonebig** [comment]
I saw PyTorch arrived, and is now slightly smaller, So I'll make a test.

**Comment by stonebig** [comment]
being only 40% behind Google after just one year of existence is a good score

**Comment by RoyiAvital** [comment]
What do you mean by small?
Why does it have to be Neural Transfer?

**Comment by stonebig** [comment]
hum ... problem: https://github.com/alexis-jacq/Pytorch-Tutorials/issues/6

**Comment by RoyiAvital** [comment]
I'm not sure but is the optimizer LBFGS?
Try using something simpler :-).

**Comment by stonebig** [comment]
yes, need yout to point me on something simple

**Comment by RoyiAvital** [comment]
The MNIST is usually the introduction task for Deep Learning:

https://github.com/pytorch/examples/blob/master/mnist/main.py

**Comment by stonebig** [comment]
yes, but it's 300% un-sexy, to the opposite of style-transfer where you can re-paint the world  in Picasso/Monet/VanGogh style.

![dancing](https://user-images.githubusercontent.com/4312421/39666215-51d5ad1e-50a0-11e8-8097-00d66c5f4113.jpg)
![stonebigcasso](https://user-images.githubusercontent.com/4312421/39666216-571e2efe-50a0-11e8-8cfe-cd4aec92cfc6.JPG)


I'm just unable to find how to save the image result of Pytorch in a file at the end of the tutorial:
- imsave doesn't work
- plt.savefig savees only blank.
==> an idea ?

**Comment by stonebig** [comment]
build b1 : 2018-05-04/5


Changes from WinPython 2018-01:

**Comment by RoyiAvital** [comment]
Well,
If the data is stored as a PytorchTensor just do the following:
```
mA = mT.cpu().numpy()
```

This will convert the tensor `mT` to Numpy array `mA`.
From there, just do as you usually do with Numpy arrays.

**Comment by stonebig** [comment]
b2 (may 27th, nearly feature-complete python-3.7) pandas-0.23.0, numba-0.38 on python-3.7.0b4

**Comment by stonebig** [comment]
build b3: 2018-06-02 (python-3.7.0b5 , python-3.7.0rc is delayed)

Changes from b2:
- debugged 3.7.0 ; numba-0.38.1, Python-3.7.0.b5, Pandas-0.22.0 **downgrade**

Areas of particular interest for testers:
- Python-3.7 should be fully stable now, some decrepited package may not make it . Winpython_checker works in Jupyterlab for all but (missing) pythonnet ("import clr")


focus for next build:
- Python-3.7.0rc1,
- holoviews-1.10.5
- pandas-0.23.1  (uncertain),
- bokeh-0.13.0 (unlikely)

 MD5                             | SHA-1                                    | SHA-256                                                          | Binary                          | Size              
---------------------------------|------------------------------------------|------------------------------------------------------------------|---------------------------------|-------------------
8b802a30f7d0eed2d940903b7c0cf4e4 | 7d1195fc311913e251b809c035b0629f9e41859c | a433bd70d616a8f0e685f9964da64a714ccfb0f5b8ccb56b5e02e61607b09232 | WinPython64-3.7.0.1Zerob3.exe   |   26 646 460 Bytes
a02f199d5c521163f8874cae95dca890 | 8821de5177195353fcfa9b65d7a23ea306dac91b | 93c51b353b34db98565f9e943b81e49dbb302894ad591f45df6635f088d3defa | WinPython32-3.7.0.1Zerob3.exe   |   25 872 001 Bytes

**Comment by stonebig** [comment]
b5

 MD5                             | SHA-1                                    | SHA-256                                                          | Binary                          | Size              
---------------------------------|------------------------------------------|------------------------------------------------------------------|---------------------------------|-------------------
45a0a02b927a475f7cdcb7d9c0509b8f | 483738706ffebe4acc63c38296f82c11a5491683 | 9ba3a989990c1382b2d1735232e1f8ac86da32bd47185af3d84bab43982cfe0f | WinPython64-3.7.0.1Zerob4.exe   |   26 827 612 Bytes
b8e222dd5b0f8a257d33b92bd55e127b | 5e29fd4ae126f2469f44e7916a6e868ff60e4b73 | 13cffe45116356fe7176d64792004ca9dc44c036fe15ce392eef982fec8629c5 | WinPython32-3.7.0.1Zerob4.exe   |   26 073 954 Bytes

**Comment by RoyiAvital** [comment]
You should call the package `PyTorch` and not `Torch`.

**Comment by stonebig** [comment]
the name of packages is the one you see per `pip list`.

**Comment by RoyiAvital** [comment]
@stonebig ,
In your package list you say the Numpy is Numpy + Intel MKL.

Yet if I do `np.__config__.show()` I get:
```
blas_mkl_info:
    libraries = ['mkl_lapack95_lp64', 'mkl_blas95_lp64', 'mkl_rt']
    library_dirs = ['C:/Program Files (x86)/IntelSWTools/compilers_and_libraries_2018/windows/mkl/lib/intel64_win']
    define_macros = [('SCIPY_MKL_H', None), ('HAVE_CBLAS', None)]
    include_dirs = ['C:/Program Files (x86)/IntelSWTools/compilers_and_libraries_2018/windows/mkl/include']
blas_opt_info:
    libraries = ['mkl_lapack95_lp64', 'mkl_blas95_lp64', 'mkl_rt']
    library_dirs = ['C:/Program Files (x86)/IntelSWTools/compilers_and_libraries_2018/windows/mkl/lib/intel64_win']
    define_macros = [('SCIPY_MKL_H', None), ('HAVE_CBLAS', None)]
    include_dirs = ['C:/Program Files (x86)/IntelSWTools/compilers_and_libraries_2018/windows/mkl/include']
lapack_mkl_info:
    libraries = ['mkl_lapack95_lp64', 'mkl_blas95_lp64', 'mkl_rt']
    library_dirs = ['C:/Program Files (x86)/IntelSWTools/compilers_and_libraries_2018/windows/mkl/lib/intel64_win']
    define_macros = [('SCIPY_MKL_H', None), ('HAVE_CBLAS', None)]
    include_dirs = ['C:/Program Files (x86)/IntelSWTools/compilers_and_libraries_2018/windows/mkl/include']
lapack_opt_info:
    libraries = ['mkl_lapack95_lp64', 'mkl_blas95_lp64', 'mkl_rt']
    library_dirs = ['C:/Program Files (x86)/IntelSWTools/compilers_and_libraries_2018/windows/mkl/lib/intel64_win']
    define_macros = [('SCIPY_MKL_H', None), ('HAVE_CBLAS', None)]
    include_dirs = ['C:/Program Files (x86)/IntelSWTools/compilers_and_libraries_2018/windows/mkl/include']
```

Which means Intel MKL isn't installed.
Could we have Numpy + MKL like in Anaconda?
It will improve performance greatly!

**Comment by stonebig** [comment]
hum, wasn't,it like that also in previous versions ?

**Comment by RoyiAvital** [comment]
I actually not sure what it means.
Could it be just a log of the configuration of the computer used to compile it?

I am not an expert on that.

**Comment by RoyiAvital** [comment]
I saw that you use [Python Extension Packages for Windows - Christoph Gohlke](https://www.lfd.uci.edu/~gohlke/pythonlibs).

Maybe it is better asking him (@cgohlke) about this ((https://github.com/cgohlke))?

**Comment by hiccup7** [comment]
Spyder v3.3.0 is released: https://github.com/spyder-ide/spyder/releases

**Comment by stonebig** [comment]
spyder-3.3.0 is there. trying to fix the last issue before rc https://github.com/bloomberg/bqplot/issues/671

**Comment by stonebig** [comment]
rc : 2018-07-08 (python-3.7.0, spyder-3.3.0)

Changes from b4:
- Python-3.7.0 final, Python-3.6.6,
- holoviews-0.10.7dev1, hvplot (renamed from holoplot), ipyleaflet-0.9.0 

Areas of particular interest for testers:
- stable and running,
- pythonnet will probably miss the python-3.7 mark

focus for next build (rc next week-end ?):
- python-3.7.0 final , python-3.6.6
- spyder-3.3.0 (with separation of spyder-kernels)

**Comment by RoyiAvital** [comment]
@stonebig , Any news on the MKL thing?

**Comment by stonebig** [comment]
no

**Comment by hiccup7** [comment]
The documentation above shows spyder_kernels v0.2.4 in the WinPython RC release.  https://github.com/spyder-ide/spyder says the minimum version of spyder_kernels is v1.0.  Want to correct for WinPython final release?

**Comment by hiccup7** [comment]
WinPython-64bit 3.6.6.1Qt5rc is working well for me so far.  I am experiencing some bugs in Spyder v3.3.0 that are planned to be fixed in v3.3.1.  Since v3.3.1 may be 4 months from now, I might update Spyder to a development version before then.  If this has problems, I want to be able restore Spyder v3.3.0.

@stonebig - Since Spyder v3.3.0 is not available at https://www.lfd.uci.edu/~gohlke/pythonlibs/ did you download wheels for Spyder and Spyder-kernels from PyPI?
